pkg/util/result/result.go

```diff
@@ -77,3 +81,47 @@ func FromError(err error) *Result {
 func TODO() error {
 	return errors.New("bailng due to error")
 }
+
+// Merge combines two results into one final result.  It properly respects all three forms of Result
+// (i.e. nil/bail/error) for both results, and combines all sensibly into a final form that represents
+// the information of both.
+func Merge(res1 *Result, res2 *Result) *Result {
+	// If both are nil, then there's no problem.  Return 'nil' to properly convey that outwards.
```

Comment:
Just a stylistic choice, but it's possible to do this very succinctly with a `switch` statement, e.g.

```go
switch {
    case res1 == nil && res2 == nil: return nil
    case res1 == nil: return res2
    case res2 == nil: return res1
    case res1.IsBail() && res2.IsBail(): return Bail()
    case res1.IsBail(): return res2
    case res2.IsBail(): return res1
    default: return FromError(multierror.Append(res1.err, res2.err))
}
```

This is fine too, up to you which syntax you prefer.